features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/info/CoordinateResolver.java

```diff
@@ -98,22 +99,21 @@ Result resolve(Collection<Integer> nodeIds) {
                 .filter(node -> !Strings.isNullOrEmpty(geoLocation(node).asAddressString()))
                 .collect(Collectors.toList());
 
-        // Retrieve coordinates
-        Map<Integer, Coordinates> nodeIdToCoordinateMapping = nodesWithAddress.stream().collect(Collectors.toMap(eachNode -> eachNode.getId(),
-                eachNode -> {
-                    final String addressString = geoLocation(eachNode).asAddressString();
-                    try {
-                        org.opennms.features.geocoder.Coordinates coordinates = geocoderService.getCoordinates(addressString);
-                        if (coordinates != null) {
-                            return new Coordinates(coordinates.getLongitude(), coordinates.getLatitude());
-                        }
-                    } catch (GeocoderException e) {
-                        LOG.warn("Couldn't resolve address '%s' for node id: %s, label: %s'", addressString, eachNode.getId(), eachNode.getLabel(), e);
-                    }
-                    return null;
-                }));
-
-        // Convert nodesWithLongLat
+        // Retrieve coordinates for nodes with a adress, but not logitude/latitude
+        Map<Integer, Coordinates> nodeIdToCoordinateMapping = new HashMap<>();
+        for (OnmsNode eachNode : nodesWithAddress) {
+            final String addressString = geoLocation(eachNode).asAddressString();
+            try {
+                org.opennms.features.geocoder.Coordinates coordinates = geocoderService.getCoordinates(addressString);
```

Comment:
The timeout will help reduce page load times, but won't eliminate the slowness all together, since we are still blocking the page while attempting to geocode each node in series.

Further, if this geocoding fails during the initial page load, it will still be reattempted every-time the view changes.

I think we need a better way of handling this.